From e764efb524fb7da327763fc85929d71537c333ca Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Tue, 16 Jan 2018 10:35:09 +0800
Subject: [PATCH] media: cec: Set kernel not do retransmit

HDMI protocol does not allow retransmit more than
five times when send cec message failed. In android
system, we do retransmit in framework only.

Change-Id: I7bd1a4aa48874be11e4948d60e0c36077268c1c9
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/media/cec/cec-adap.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/cec/cec-adap.c b/drivers/media/cec/cec-adap.c
index da218401b8fa..2e66ee402801 100644
--- a/drivers/media/cec/cec-adap.c
+++ b/drivers/media/cec/cec-adap.c
@@ -495,8 +495,11 @@ int cec_thread_func(void *_adap)
 		if (data->msg.len == 1 && adap->is_configured)
 			attempts = 2;
 		else
+#ifdef CONFIG_ANDROID
+			attempts = 1;
+#else
 			attempts = 4;
-
+#endif
 		/* Set the suggested signal free time */
 		if (data->attempts) {
 			/* should be >= 3 data bit periods for a retry */
-- 
2.35.3

